source/common/upstream/load_balancer_impl.h

```diff
@@ -70,10 +73,66 @@ class ZoneAwareLoadBalancerBase : public LoadBalancerBase {
                             const envoy::api::v2::Cluster::CommonLbConfig& common_config);
   ~ZoneAwareLoadBalancerBase();
 
+  // When deciding which hosts to use on an LB decision, we need to know how to index into the
+  // priority_set. This priority_set cursor is used by ZoneAwareLoadBalancerBase subclasses, e.g.
+  // RoundRobinLoadBalancer, to index into auxillary data structures specific to the LB for
+  // a given host set selection.
+  struct HostsSource {
+    enum class SourceType {
+      // All hosts in the host set.
+      AllHosts,
+      // All healthy hosts in the host set.
+      HealthyHosts,
+      // Healthy hosts for locality @ locality_index.
+      LocalityHealthyHosts,
+    };
+
+    HostsSource() {}
```

Comment:
nit: do we need default constructor?

Reply:
Yeah, it's a bit more convenient to build it incrementally in load_balancer_impl.cc with this.